### PR TITLE
Fix k8s deployment

### PR DESCRIPTION
**Pull request by jfinkhaeuser**
Work in progress - it seems to be mainly documentation changes at this point.

### CODE REVIEW

**Comment by jfinkhaeuser** [comment]
TODOs:

- [ ] `docker-engine` repository no longer exists; it shouldn't be configured.
- [ ] `docker-ce` and `docker-ce-cli` are installed in conflicting versions; should pin to `18.09` (there exists `19.x`, but kubespray does not officially support that.

**Comment by jfinkhaeuser** [comment]
https://status.quay.io/incidents/qdn3wz80kzww - quay.io is seeing error rates, preventing the installation of this image.

**Comment by jfinkhaeuser** [comment]
https://proseo-registry.eoc.dlr.de/ - proseo registry is erroring out

**Comment by jfinkhaeuser** [comment]
So, with `download_validate_certs` set to `False`, a majority of the k8s setup runs through. But some service isn't running yet. I'll look into that tomorrow.

**Comment by jfinkhaeuser** [comment]
`kube-apiserver` was somehow listening on `10251`, preventing `kube-scheduler` to run - killing both and letting k8s restart them seems to have done the trick. There probably is a missing restart after configuring `kube-apiserver` somewhere?

**Comment by jfinkhaeuser** [comment]
Ok, what seems to be the last failure relates to the node not joining the cluster, because they can't connect to the master's port 6443 - there is a service listening there, and locally connecting to it is fine. Also pinging the master works, meaning it looks like an OTC networking issue. Looking into that.

**Comment by jfinkhaeuser** [comment]
TL;DR is, firewalld isn't configured to work with k8s, so would either need ports opened, or be disabled.

**Comment by jfinkhaeuser** [comment]
There's a certbot issue in the postinstall script; the k8s cluster is up and running, but the bastion doesn't yet have its nginx proxy configured due to the certbot issue.

**Comment by jfinkhaeuser** [comment]
So for the certbot issue(s), we'll need to discuss, I think:
- Using ansible to configure nginx for certbot usage requires certbot to already have successfully generated the certificates, which is not part of the ansible script -> there is a bootstrapping problem
- Currently, proseo-k8s-gate.de does not point at the bastion host, so certbot won't succeed with that domain name.
